common/stats/stats.h

```diff
@@ -149,6 +151,9 @@ class Stats {
   // ostrich library dumps stats).
   std::string DumpStatsAsText();
 
+  // Return stats as folly::dynamic.
+  folly::dynamic DumpStatsAsDyn();
```

Comment:
If we really need json formated stats, I think we should return string here, and rename the function to DumpStatsAsJson(). and remove folly/json.h folly/dynamic.h from stats.h and status_server.h, but to put the "constructing dynamic" and "convert dynamic to json" logic in stats.cpp.